src/lib/eliom_client.client.mli

```diff
@@ -315,6 +327,59 @@ val init : unit -> unit
 
 val set_reload_function : (unit -> unit -> unit Lwt.t) -> unit
 
+(** Store the document/body of the current page, which will be used when going 
+    back to this page in history. 
+    A typical use case of this function is storing the dom when leaving 
+    a page. i.e. [Eliom_client.onchangepage Eliom_client.push_history_dom ]
+*)
+val push_history_dom : unit -> unit
+
+(** Install an onchangepage handler for the current page. When leaving this page,
+    the function [push_history_dom] will be called. If history_changepage_handler
+    is not null, it will be called with the current state id as the first parameter. 
+*)
+val install_history_changepage_handler : unit -> unit
+
+(** Why do we need this function ?
+    Suppose that we want to have a page transition on coming back from details 
+    to list page, we need usually a parameter (e.g. a screenshot of the list 
+    page) to realize the animation of page transition and the parameter is 
+    generally created just before we leave the list page in order to store the
+    latest information (such as the scroll position). Normally, in order to do this, 
+    we register an onchangepage handler in the beginning of the handler of the service 
+    that generates the list page. However, when we reload the list page by replacing 
+    the current document/body with the one stored in cache, the handler will be lost.
+    So we need a specific function to re-register the handler everytime the list
+    page is reloaded from cache.    
+
+    Suppose also that we have a hashtable which maps a state_id into a parameter 
+    (e.g. a screenshot) corresponding to the page characterized by the state_id. 
+
+    [set_history_changepage_handler f] registers an onchangepage handler for 
+    the all pages reloaded from cache : [Eliom_client.onchangepage (f id)]. 
+    Here f should take a state id as argument so that it can register a particular 
+    handler for the page associated with the state_id.
+
+    Notice that normally you need to call this function before installing the 
+    changepage handler for any specific page, in other words, before calling 
+    any [install_history_changepage_handler ()], so that the function 
+    [install_history_changepage_handler] will execute the handler function 
+    before leaving the page for the first time.
+*)
+val set_history_changepage_handler : (int -> onchangepage_event -> unit) -> unit
+
+(** [set_animation_function f] sets the animation function which is used to
+    realize the page transition on coming back in history. With the same
```

Comment:
to only when coming back but also when going forward / clicking on any link, n'est-ce pas ?